src/txn/txn.c

```diff
@@ -124,9 +124,10 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     txn_global = &S2C(session)->txn_global;
     txn_shared = WT_SESSION_TXN_SHARED(session);
 
-    WT_ASSERT(session,
+    WT_ASSERT_OPTIONAL(session,
       txn_shared->pinned_id == WT_TXN_NONE || session->txn->isolation == WT_ISO_READ_UNCOMMITTED ||
-        !__wt_txn_visible_all(session, txn_shared->pinned_id, WT_TS_NONE));
+        !__wt_txn_visible_all(session, txn_shared->pinned_id, WT_TS_NONE),
+      "Releasing snapshot that is not globally visible");
```

Comment:
This assertion confuses me, I don't think it should be optional but I guess its an "expensive" one. That said you should be able to release a snapshot at any time regardless of global visibility so I don't think that's what this is checking?

Reply:
Per our discussion with Michael, I think we can update this error to be: "A transactions pinned id cannot become globally visible before the snapshot is released."